### PR TITLE
Nigeria refactor

### PR DESCRIPTION
**Pull request by camappel**
Feel free to contribute at #2099

### CODE REVIEW

File: scripts/scripts/testing/automations/incremental/nigeria.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+def main():
+    data = pd.read_csv("automated_sheets/Nigeria.csv").sort_values(by="Date", ascending=False)
+
+    headers = {
+        "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36"
+    }
+    source_url = "http://covid19.ncdc.gov.ng/"
+
+    req = requests.get(source_url, headers=headers)
+    soup = BeautifulSoup(req.content, "html.parser")
```
Comment (lucasrodes) [inline scripts/scripts/testing/automations/incremental/nigeria.py:16]: You can try to use `cowidev` library here:

```python
soup = get_soup(source_url)
```

Note that for this to work you need to import `get_soup` function at the top of the script as:

```python
from cowidev.utils.web import get_soup
```

File: scripts/scripts/testing/automations/incremental/nigeria.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import pandas as pd
+import requests
+from bs4 import BeautifulSoup
+from datetime import date
+
+
+def main():
+    data = pd.read_csv("automated_sheets/Nigeria.csv").sort_values(by="Date", ascending=False)
```
Comment (lucasrodes) [inline scripts/scripts/testing/automations/incremental/nigeria.py:8]: To ensure the path works regardless of where the script is executed, I'd suggest using function `get_project_dir` as:

```python
path = os.path.join(get_project_dir(), "scripts", "scripts", "testing", "automated_sheets", "Nigeria.csv")
data = pd.read_csv(path).sort_values(by="Date", ascending=False)
```

Note you need to import:

```python
import os
from cowidev.utils.utils import get_project_dir
```

File: scripts/scripts/testing/automations/incremental/nigeria.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                "Cumulative total": cumulative_total,
+                "Country": "Nigeria",
+                "Units": "samples tested",
+                "Source URL": source_url,
+                "Source label": "Nigeria Centre for Disease Control",
+            }
+        )
+
+        df = pd.concat([new, data], sort=False)
+        df.to_csv("automated_sheets/Nigeria.csv", index=False)
```
Comment (lucasrodes) [inline scripts/scripts/testing/automations/incremental/nigeria.py:34]: Use

```python
df.to_csv(path, index=False)
```

**Review by lucasrodes — CHANGES_REQUESTED** [review]

**Review by camappel — COMMENTED** [review]
Thank you for incorporating the cowidev library

**Review by lucasrodes — CHANGES_REQUESTED** [review]

File: scripts/scripts/testing/automations/incremental/nigeria.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+def main():
+    path = os.path.join(get_project_dir(), "scripts", "scripts", "testing", "automated_sheets", "Nigeria.csv")
+    data = pd.read_csv(path).sort_values(by="Date", ascending=False)
+
+    source_url = "http://covid19.ncdc.gov.ng/"
+
+    soup = get_soup(source_url)
+
+    cumulative_total = int(soup.find("div", class_="col-xl-3").find("span").text.replace(",", ""))
```
Comment (lucasrodes) [inline scripts/scripts/testing/automations/incremental/nigeria.py:16]: Use [clean_count()](https://github.com/owid/covid-19-data/blob/a3a100d280ecbf0e5da0c654b4d09fae067da0a3/scripts/src/cowidev/utils/clean/numbers.py#L88)

```python
from cowidev.utils.clean import clean_count
element = soup.find("div", class_="col-xl-3").find("span")
cumulative_total = clean_count(element.text)
```

**Review by lucasrodes — CHANGES_REQUESTED** [review]

**Pull request merged**